internal/ports.go

```diff
@@ -0,0 +1,7 @@
+package internal
+
+type PortAssigner interface {
```

Comment:
[nit] Would be beneficial to have a comment for the exported interfaces e.g., `PortAssigner`.

```suggestion
// PortAssigner assigns unique network ports for node instances and tests.
type PortAssigner interface {
```